### ISSUE TITLE
i18n locale won't be passed to custom dev server

### ISSUE THREAD

**Issue by HamAndRock** [vercel/next.js#24215]
### What version of Next.js are you using?

10.1.3

### What version of Node.js are you using?

14.15.1

### What browser are you using?

Chrome

### What operating system are you using?

Windows 10

### How are you deploying your application?

next start

### Describe the Bug

I am using custom server, described at [https://nextjs.org/docs/advanced-features/custom-server](nextjs docs). And also using the i18n routing recetly introduced in nextjs.

Hovewer when trying read `locale` property in Router (dev mode), the value is undefined. There is no mention on how to fix it in docs. It works flawlessy in production but I have to prepend i18n slug before template route, otherwise 404 is rendering in production (again no mention in the docs), when using this approach (prepanding i18n slug before template path) in dev mod it renders 404.

### Expected Behavior

I would expect the locale property to be filled and same behaviour in dev and prod enviroment.

### To Reproduce

Create simple custom server like desribed in docs. Add i18n configuration in next config with locale `['cz-cs']`, it doesn't work it others too.

Try to render template.jsx file by entering `/cz-cs/index2` in browser in dev mode. If the production check wasn't in place it would render 404 in dev
![image](https://user-images.githubusercontent.com/24235083/115222346-eba2ec00-a10a-11eb-8129-afb33cd8cb77.png)

server.js
```js
const { createServer } = require('http');
const { parse } = require('url');
const next = require('next');

const dev = process.env.NODE_ENV !== 'production';
const app = next({ dev });
const handle = app.getRequestHandler();
const routing = require('./src/utils/routing');

const routes = routing.routes;

app.prepare().then(() => {
  createServer((req, res) => {
    const parsedUrl = parse(req.url, true);
    let { pathname } = parsedUrl;

    const isProduction = process.env.NODE_ENV === 'production';
    

    if (pathname.includes('index2')) {
      if (isProduction) app.render(req, res, '/cz-cs/template'); // prevents from 404 rendering in production
      else app.render(req, res, '/template'); // if I add cz-cs here it will render 404 in dev
    }

    //default
    else {
      handle(req, res, parsedUrl);
    }
  }).listen(3000, err => {
    if (err) throw err;
    console.log('> Ready on http://localhost:3000');
  });
});


```

template.jsx
```js
import { useRouter } from 'next/router';

const Template = () => {
  const router = useRouter();

  console.log(router); // will log locale = undefined in dev and cz-cs in production

  return <div></div>;
};

export default Template;
```

**Comment by HamAndRock** [comment]
@ijjk Hey the duplicate id is same as id of this issue..

**Comment by ijjk** [comment]
Good catch, updated with the correct issue https://github.com/vercel/next.js/issues/20761

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
